### ISSUE TITLE
nice stack trace handling

### ISSUE THREAD

**Issue by davidmarkclements** [senecajs/seneca#98]
If there is a try/catch in core, and all it does is print a trace via seneca log then exit, we could avoid the overhead of a try catch by using Error.prepareStackTrace

```
Error.prepareStackTrace = function (error, stack) {
  seneca.error.log(error, JSON.stringify(stack))
  return ''; //return empty string to overwrite native stack trace
}

```

Also, we could do really nice coloured stack traces (generic this time:)

```
var util = require('util');
var colors = require('colors')
Error.prepareStackTrace = function (error, stack) {
  return stack.map(function (frame) {
    var fn = frame.getFunction();
    var line = frame.getLineNumber();
    var fnId = frame.getFunctionName() || 
      frame.getMethodName() || 
        ((fn+'').split('{')[0].trim() +
        ' { [body] }') 
   
    return [ 
      error.red, 
      '\n',
      frame.getFileName().cyan, (' ' + line + ' ').bgYellow.black,
      (' ' + fnId + ' ').gray
      ].join(' ')
  }).join('\n');
}

throw x;

```

[Trello Card](https://trello.com/c/6iUU0Wgf/85-seneca-nice-stack-trace-handling)

**Comment by juliangruber** [comment]
patching globals likely leads to confusion and bugs, i'm -1

**Comment by rjrodger** [comment]
Yeah, agreed @juliangruber this seems like asking for trouble down the road.
@davidmarkclements leaving this issue here as it may yet be interesting

**Comment by davidmarkclements** [comment]
Hold on a sec, we're confusing notions here. You don't patch globals - e.g. say Object.prototype because it may clash with others doing the same and so forth. `Error.prepareStackTrace` is a supported V8 API - it's specifically there to be overwritten for custom stack traces - see here https://code.google.com/p/v8-wiki/wiki/JavaScriptStackTraceApi 

The taboo of patching globals most definitely doesn't apply here.

**Comment by davidmarkclements** [comment]
oh btw, turned this into a module http://npmjs.org/cute-stack

**Comment by mcdonnelldean** [comment]
@davidmarkclements @mcollina Might be worth the two of ye talking about this. Merged with #398
